parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -2785,6 +2785,12 @@ opt_block_args_tail:
                         lambda->end);
                       DIAGCHECK();
```

Comment:
I'm usually skeptical of this line, I have no clue what it does nor what would happen to the parser if we deleted them all. It usually seems to prevent me from doing the sorts of errors handling that I'm trying to do 🤔.

Reply:
Reading through it, it seems like the purpose is to make it so that if any diagnostics were recorded as errors while the action attached to the production was executing, `YYERROR` will be triggered. I wonder if we should be handling that sort of thing more generally via error nodes?